pkg/networkserver/downlink.go

```diff
@@ -249,21 +236,14 @@ func (ns *NetworkServer) generateDownlink(ctx context.Context, dev *ttnpb.EndDev
 			if err != nil {
 				return nil, generateDownlinkState{}, errEncodeMAC.WithCause(err)
 			}
-			if mType == ttnpb.MType_UNCONFIRMED_DOWN &&
```

Comment:
`ExpectAnswer` is redundant, since NS can just use `PendingRequests` slice for this purpose.
Also, adapt for class B, move this closer to where it's actually used and out of the loop.